server/src/main/java/io/crate/execution/dml/upsert/TransportShardUpsertAction.java

```diff
@@ -284,68 +281,23 @@ private static String getExceptionMessage(Throwable e) {
         return message != null ? message : e.getClass().getName();
     }
 
-    private WriteReplicaResult<ShardUpsertRequest> legacyReplicaOp(IndexShard indexShard, ShardUpsertRequest request) throws IOException {
-        boolean traceEnabled = logger.isTraceEnabled();
-        Translog.Location location = null;
-        for (ShardUpsertRequest.Item item : request.items()) {
-            if (item.seqNo() == SequenceNumbers.SKIP_ON_REPLICA || item.source() == null) {
-                if (traceEnabled) {
-                    logger.trace(
-                        "[{} (R)] Document with id={}, marked as skip_on_replica",
-                        indexShard.shardId(),
-                        item.id()
-                    );
-                }
-                continue;
+    private static boolean shouldIgnoreAllItems(ShardUpsertRequest req) {
+        for (ShardUpsertRequest.Item item : req.items()) {
```

Comment:
We create requests with `null` insert columns in two places, both of which use no-op items, so while the old code used the `legacyReplicaOp` method, it didn't actually do any doc parsing.  This assertion double checks that this still holds.